pkg/catalog/catalog.go

```diff
@@ -84,20 +90,29 @@ func (c *Catalog) GetWithVersion(sourceName, profileName, profileVersion string)
 
 type profileDescriptionWithVersion struct {
 	profileDescription profilesv1.ProfileDescription
-	semverVersion      *version.Version
+	semverVersion      *semver.Version
 }
 
-func getLatestVersion(profiles []profilesv1.ProfileDescription, profileName string) *profilesv1.ProfileDescription {
+// GetGreaterThan returns all profiles which are of a greater version for a given profile with a version.
+func (c *Catalog) GetGreaterThan(sourceName, profileName, profileVersion string) []profilesv1.ProfileDescription {
 	var profilesWithValidVersion []profileDescriptionWithVersion
-
-	for _, p := range profiles {
-		v, err := version.NewVersion(p.Version)
+	profiles, ok := c.m.Load(sourceName)
+	if !ok {
+		return nil
+	}
+	cv, err := version.ParseVersion(profileVersion)
+	if err != nil && profileVersion != "latest" {
+		return nil
+	}
+	for _, p := range profiles.([]profilesv1.ProfileDescription) {
+		v, err := version.ParseVersion(p.Version)
 		if err != nil {
 			continue
 		}
-
 		if p.Name == profileName {
-			profilesWithValidVersion = append(profilesWithValidVersion, profileDescriptionWithVersion{profileDescription: p, semverVersion: v})
+			if profileVersion == "latest" || (cv != nil && v.GreaterThan(cv)) {
```

Comment:
do we need the nil check? looks smelly :laughing:

Reply:
We don't necessarily need it, but the static check is happier this way. :)

That's because the latest check above might turn the cv into a nil pointer. But that wouldn't be used anyways, since profileVersion == "latest"... But the static checker is still complaining about it.

Reply:
Oh no, actually I changed this recently and didn't update. :D :D